src/test/java/sg/dex/starfish/impl/TestResources.java

```diff
@@ -13,12 +14,43 @@
 public class TestResources {
 
 	@Test public void testResourceAsset() {
-		Asset ua=ResourceAsset.create("{}","assets/hello.txt");
+		Asset ua=ResourceAsset.create("assets/hello.txt","{}",false);
 		byte[] bs=ua.getContent();
 		String s=new String(bs,StandardCharsets.UTF_8);
 		assertEquals("Hello Starfish",s);
 
 		assertNotNull(ua.getMetadataString());
 		assertNotNull(((ResourceAsset) ua).getContentSize());
 	}
+
+	@Test public void testResourceAssetWithContetHash() {
+		Asset ua=ResourceAsset.create("assets/hello.txt","{}",true);
+		byte[] bs=ua.getContent();
+		String s=new String(bs,StandardCharsets.UTF_8);
+		assertEquals("Hello Starfish",s);
+		assertNotNull(ua.getMetadata().get(Constant.CONTENT_HASH));
+
+		assertNotNull(ua.getMetadataString());
+		assertNotNull(((ResourceAsset) ua).getContentSize());
+	}
+	@Test public void testResourceAssetWithoutContetHash() {
+		Asset ua=ResourceAsset.create("assets/hello.txt","{}",true);
+		byte[] bs=ua.getContent();
+		String s=new String(bs,StandardCharsets.UTF_8);
+		assertEquals("Hello Starfish",s);
+		assertNotNull(ua.getMetadata().get(Constant.CONTENT_HASH));
```

Comment:
Use asssertEquals in places like this and check that the content hash is correct!